app/controllers/UploadLetterOfAuthorityController.scala

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import javax.inject.{Inject, Singleton}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import play.api.Configuration
+import play.api.i18n.{I18nSupport, Messages, MessagesApi}
+import play.api.mvc._
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+
+import config.FrontendAppConfig
+import controllers.actions.{DataRequiredAction, DataRetrievalActionProvider, IdentifierAction}
+import models._
+import navigation.Navigator
+import pages._
+import services.fileupload.FileService
+import views.html.UploadLetterOfAuthorityView
+
+@Singleton
+class UploadLetterOfAuthorityController @Inject() (
+  override val messagesApi: MessagesApi,
+  override val controllerComponents: MessagesControllerComponents,
+  identify: IdentifierAction,
+  getData: DataRetrievalActionProvider,
+  requireData: DataRequiredAction,
+  view: UploadLetterOfAuthorityView,
+  fileService: FileService,
+  navigator: Navigator,
+  configuration: Configuration,
+  appConfig: FrontendAppConfig
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  private val mode: Mode                       = NormalMode // TODO: allow other modes other than NormalMode.
```

Comment:
Is this something you wanted to fix before the PR is merged? In other controllers the mode is passed in as a parameter for onPageLoad()

Reply:
I tried passing it in but was getting compilation errors (incorrect number of arguments). I can raise another story/bug for this.

Reply:
I'm happy to resolve this comment as we don't need this piece of functionality until the agent creds journey is completed all the way up to the 'check your answers' page. We have to remember once we get there that this flag needs to be implemented as a parameter, not hard coded.